### PR TITLE
HIVE-23871: ObjectStore should properly handle MicroManaged Table props

### PR DESCRIPTION
**Pull request by pgaref**
ObjectStore should properly handle MicroManaged Table properties

Change-Id: Ia5db047419a11504f3c6047a1eb63acd2a14bdc3

## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HIVE-XXXXX: Fix a typo in YYY)
For more details, please see https://cwiki.apache.org/confluence/display/Hive/HowToContribute

### CODE REVIEW

File: data/files/table1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+1	Acura	4
```
Comment (mustafaiman) [inline data/files/table1:1]: can we give the file a non generic name?

File: ql/src/test/results/clientpositive/llap/load_micromanaged_delim.q.out [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+#### A masked pattern was here ####
+POSTHOOK: query: SELECT * FROM delim_table_trans
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@delim_table_trans
+#### A masked pattern was here ####
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	NULL
```
Comment (mustafaiman) [inline ql/src/test/results/clientpositive/llap/load_micromanaged_delim.q.out:186]: I did not get the behavior of the last two tables. Why do they print all NULLs?

File: ql/src/test/queries/clientpositive/load_micromanaged_delim.q [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+dfs -cp ${system:hive.root}/data/files/table1 ${system:test.tmp.dir}/delim_table_trans/;
+
+-- Checking that MicroManged and External tables have the same behaviour with delimited input files
+-- External table
+CREATE EXTERNAL TABLE delim_table_ext(id INT, name STRING, safety INT) ROW FORMAT DELIMITED FIELDS TERMINATED BY '\t' STORED AS TEXTFILE LOCATION '${system:test.tmp.dir}/delim_table_ext/';
+describe formatted delim_table_ext;
+SELECT * FROM delim_table_ext;
+
+-- SET hive.create.as.acid=true
+-- SET hive.create.as.insert.only=true
```
[the remark this answers is not in the document]
Comment (pgaref) [reply ql/src/test/queries/clientpositive/load_micromanaged_delim.q]: Creates the same behaviour as the Table properties below but I agree it makes sense to remove it

File: data/files/table1 (hunk 1 above)
Comment (pgaref) [reply data/files/table1]: sure, done

File: ql/src/test/results/clientpositive/llap/load_micromanaged_delim.q.out (hunk 2 above)
Comment (pgaref) [reply ql/src/test/results/clientpositive/llap/load_micromanaged_delim.q.out]: This is exactly the behavior this patch fixes! Apparently I run the test only before the changes -- this is now updated

**Comment by pgaref** [comment]
Thanks for the review @mustafaiman ! Addressed your comments as part of the second commit -- I am still expecting some q.out difference so I am currently waiting for the qtests to run

**Pull request closed without merging**